File: admin-frontend/src/components/participationStatus.ts
export const PARTICIPATION_STATUSES = [
  "BEFORE_PARTICIPATION",
  "PARTICIPATING",
  "DROPPED",
] as const;
export type ParticipationStatus = (typeof PARTICIPATION_STATUSES)[number];

export const PARTICIPATION_STATUS_LABELS: Record<ParticipationStatus, string> =
  {
    BEFORE_PARTICIPATION: "参画前",
    PARTICIPATING: "参画中",
    DROPPED: "参画落ち",
  };

export const PARTICIPATION_STATUS_OPTIONS: {
  value: ParticipationStatus;
  label: string;
}[] = [
  {
    value: "BEFORE_PARTICIPATION",
    label: PARTICIPATION_STATUS_LABELS.BEFORE_PARTICIPATION,
  },
  {
    value: "PARTICIPATING",
    label: PARTICIPATION_STATUS_LABELS.PARTICIPATING,
  },
];

export function getParticipationStatusLabel(value: string): string {
  return PARTICIPATION_STATUS_LABELS[value as ParticipationStatus] ?? value;
}

File: backend/common/src/main/kotlin/com/mametosho/domain/model/shared/ParticipationStatus.kt
package com.mametosho.domain.model.shared

/**
 * 参画ステータス。
 *
 * BEFORE_PARTICIPATION（参画前）→ PARTICIPATING（参画中）→ DROPPED（参画落ち）の直線遷移のみ許可。
 * DROPPED になると以降のステータス変更は不可。
 * PARTICIPATING の店舗のみ CS（一般ユーザー向け）に公開される。
 */
enum class ParticipationStatus {
    BEFORE_PARTICIPATION,
    PARTICIPATING,
    DROPPED,
}
